rust/connlib/tunnel/src/lib.rs

```diff
@@ -200,6 +203,34 @@ where
 
     pub fn poll_next_event(&self, cx: &mut Context<'_>) -> Poll<Event<TRoleState::Id>> {
         loop {
+            if let Some(conn_id) = self.peers_to_stop.lock().pop_front() {
```

Comment:
To be sure this works without awaking the loop because this will always be immediatley `Some` after [this](https://github.com/firezone/firezone/pull/2510/files#diff-0162754cdc268cd24fa9b67f5fe425339a80a03e4d188d075f4351ddeed8476eR326)?

Reply:
What do you mean by awaking the loop? We will immediately stop the peer after we've pushed it because we are popping it off here.

Does that make sense?